File: Packages/CodeText/Tests/CodeTextTests/Samples/sample.go
// Package till is a small shop's till, for the colour samples (041).
package till

import (
	"errors"
	"fmt"
)

const Float = 150.0
const mask = 0xFF

// Receipt is one sale.
type Receipt struct {
	Item  string
	Total float64
	Count int
}

type Till struct {
	float    float64
	receipts []Receipt
}

var ErrNothing = errors.New("cannot sell nothing")

func (t *Till) Sell(item string, price float64, count int) (Receipt, error) {
	if count <= 0 {
		return Receipt{}, fmt.Errorf("sell %q: %w", item, ErrNothing)
	}
	r := Receipt{Item: item, Total: price * float64(count), Count: count}
	t.receipts = append(t.receipts, r)
	t.float += r.Total
	return r, nil
}

func (t *Till) Takings() (sum float64) {
	for _, r := range t.receipts {
		sum += r.Total
	}
	return
}

File: Packages/CodeText/Tests/CodeTextTests/Samples/sample.rb
# A small shop's till, for the colour samples (041).
require "bigdecimal"

FLOAT = BigDecimal("150.00")
MASK = 0xFF

Receipt = Struct.new(:item, :total, :count)

class Till
  attr_reader :receipts

  def initialize(float = FLOAT)
    @float = float
    @receipts = []
  end

  # Sells `count` of an item.
  def sell(item, price, count: 1)
    raise ArgumentError, "cannot sell #{count} of #{item}" if count <= 0

    receipt = Receipt.new(item, price * count, count)
    @receipts << receipt
    @float += receipt.total
    receipt
  end

  def takings = @receipts.sum(&:total)
end

till = Till.new
till.sell("coffee", BigDecimal("3.20"), count: 2)
puts till.takings, :done, nil, true, 1.5e3
